src/lab3/types/Transformable.ts

```diff
@@ -0,0 +1,17 @@
+import { Traceable } from '../../lab1/types/Traceable';
+
+export interface Rotatable {
+  rotate(angleX: number, angleY: number, angleZ: number): void;
+}
+
+export interface Translatable {
+  translate(x: number, y: number, z: number): void;
+}
+
+export interface Scalable {
+  scale(x: number, y: number, z: number): void;
+}
+
+export interface Transformable extends Rotatable, Translatable, Scalable {}
+
+export interface SceneObject extends Transformable, Traceable {}
```

Comment:
Better remove this interface, and make all classes with `implements Traceable, Transformable`. It will be more clear to determine object that can be rendered or transformed

Reply:
It's just a shortcut. For me it's more convenient to write this (now `TraceableTransformable` than Transformable, Traceable or Transformable & Traceable)